### DISCUSSION TITLE
Students cannot accept assignments: the visbility [sic] of the starter code is out of sync with the classroom.

### DISCUSSION THREAD

**Discussion by VictorNorman** [community/community#131139] [GitHub Education] [answered]
### Hi everyone,

I decided to create my Fall Classroom today for my CS112 course. I created a new org, and then did Reuse Assignment on an old copy of the course. I imported all my assignments. Now, I'm trying to accept an assignment (as if I were a student).

It says it is creating the new assignment for me, please refresh, etc... then I get this error every time:

> There was an error accepting the assignment. GitHub repository could not be created. The visbility of the starter code is out of sync with classroom. Your instructor may need to re-create the assignment or change the visibililty of the starter code repository.. If this problem persists, please have the Classroom admin contact support.
> repository
> 
> You accepted the assignment, Lab0 but something went wrong. We have not received any updates for this assignment. You may want to reaccept the assignment. 

I'm the instructor and I don't know how to fix this... What does "change the visibililty [sic]" mean? 

My repo (https://github.com/24fa-cs112-org/cs112-lab0-assignment) was a Private Template. Now I've changed it to a Public Template. That does not seem to have helped.

**Comment by JessRudder** [comment]
Hi @VictorNorman. I'm sorry the copy feature is not working as expected. If I understand correctly, you got the error message saying the visibility was out of sync and then you changed the visibility of the starter code repo to public, is that correct?

If that's the case, the first step of me helping to debug is going to seem a bit silly as I'd like for you to change the visibility of the starter code repo back to private. The reason for this is that the visibility on the assignment is set to private and that needs to match the starter code. I understand that's what you had before that wasn't working, but, I need to get it back into the "right" state so we can continue debugging what's off.

To make sure you're looking at the right starter code repository, please view the assignment in Classroom. Under the assignment title should be a line that says something like "Starter code from org-name/repo-name". The org/repo name should be a link that will take you to the proper repository in GitHub.

Once the starter code repo is set to private visibility again, please try to accept it again (that will give me an updated error to look into) and then let me know in this thread.

**Reply by SimonSchillberg** [reply]
Exactly same Problem here. 
The Repository is marked as Template Repository with visibility set to private.

Changing visibility for repository from Private to Public has no effect. Error message does not change.

Option to change visibility of Classroom/Assignment does not exist (as far as i checked) 

Changing visibility of the repository back from Public to Private results in same error message when trying to accept the assignment:

> There was an error accepting the assignment. GitHub repository could not be created. The visbility of the starter code is out of sync with classroom. Your instructor may need to re-create the assignment or change the visibililty of the starter code repository.. If this problem persists, please have the Classroom admin contact support.

**Reply by VictorNorman** [reply]
OK. I've changed the starter repo back to a private template.

![Screenshot 2024-07-08 at 9 46 04 AM](https://github.com/community/community/assets/6807900/b71524ec-298c-49e4-bbe9-e9fddc69b382)

Here is the Lab0 assignment setup screen:

![Screenshot 2024-07-08 at 9 46 35 AM](https://github.com/community/community/assets/6807900/f309a3d2-0e90-4493-abe7-efd5fc3da778)

I just tried to re-accept the assignment and still get the same error.

**Reply by GSibinger** [reply]
The issue seems to be this: 
For the new feature, where a repository is created from a template repo from which student repos are then forked, you apparently need to create new assignments. This initial - and fully squashed - repository seems to be created from the provided template repo at the moment an assignment is created, and only then.
Student repositories can only be forked from repositories with a single commit, following a recent update from probably last week.

In my team we noticed that behavior yesterday. We had exactly one assignment that we could still create the student repos from. The main difference this one had to all the others was that it had only exactly one commit.

So, now we squashed all the other templates to a single commit, because re-creating every single assignment simply wasn't an option. With these squashed templates we can create student repos again. They are forked directly from the template, though.

So it seems there's these two options right now: 
1. re-create every single assignment, so that the initial repo is created from the template
2. squash the template repo and have the new repos be forked directly from the template repo

I am not happy with either solution, to be honest.

**Reply by JessRudder** [reply]
@VictorNorman and @SimonSchillberg we believe we have identified the issue that's preventing assignment acceptance for your assignments and hope to have a solution deployed tomorrow. I'll update here when it is out. Thank you so much for the additional information, it was very helpful for debugging.

@GSibinger you shouldn't need to squash any commits in order for a repository to work as starter code. The starter code repo does need to be set as a template repository though. If you have an example assignment where the starter code is a template repo and it's still not working for you, please let me know the name of the classroom and assignment and the team will look into it.

**Reply by GSibinger** [reply]
Hi @JessRudder, thank you for your response.

All our assignments are working now because we squashed the template repositories. While there might have been a configuration issue, we couldn't identify or fix it.

The starter code repos were all set as private templates. We verified this on July 1st when your changes were rolled out. We could create student repos, but we did not notice at the time that they were forked directly from the template. On Monday, a student reported that they could no longer take assignments, encountering the same error @VictorNorman posted.

After analyzing the classrooms and repos, we re-created one assignment and noticed that the starter code repo was created correctly from our template and attached to the assignment. This did not happen for existing assignments.

You are welcome to review one of our classrooms [here](https://classroom.github.com/classrooms/65162677-php-grundlagen/assignments/gaestebuch). The template repository now has 2 branches: the main branch is the squashed one, and we renamed the original main to "develop".

Thank you for looking into this.

**Comment by twennekers** [comment]
I am having the same problem. 
Time is running  for this referral assignment and the students are not happy.

**Comment by twennekers** [comment]
Problem persists
Reset visibility to private 
Created a new student github account 
The repository cannot be created for this student.
Reset visibility to public again. 
Old error reappeared - repository out of sync 

When I set up the assignment I actually created a repository using the link myself under the same account as the template repository was created. This worked and the repository copy is fine. 

I believe in the meantime some procedures to setup assignments on Github / Classroom have changed. Perhaps that messed something up?

**Comment by JessRudder** [comment, marked as answer]
@VictorNorman, @SimonSchillberg, @twennekers and @GSibinger we have deployed a fix that should resolve the "visibility mismatch" issue you were seeing with assignment acceptance. If the issue persists (or you are seeing other issues with assignment acceptance), please let us know (including the name of the classroom and assignment, where possible to make troubleshooting easier).

**Reply by alanarnholt** [reply]
I am having the same issue one month after your answer.  

Classroom: https://classroom.github.com/classrooms/176928420-stt3850-fall2024

Assignment:  https://classroom.github.com/a/jqzubXZu

**Reply by allenjas** [reply]
I am having the same issue in August 2024.  My students get the error "There was an error accepting the assignment. GitHub repository could not be created. The visibility of the starter code is out of sync with classroom. Your instructor may need to re-create the assignment or change the visibility of the starter code repository.. If this problem persists, please have the Classroom admin contact support." 
for:
Classroom:  https://classroom.github.com/classrooms/120423013-nku-bis-305  
Assignment example (I expect they are all broke):  https://classroom.github.com/a/U_8GgCuJ 

I don't have an issue getting to the assignments but at least 2 students have reported that they can't access the assignment and receive that error message above

**Reply by DrErickson** [reply]
I'm also getting this error for one of my students

Classroom: https://classroom.github.com/classrooms/77716313-csueb-cs301-01-fall-2024
Assignment: https://classroom.github.com/a/CNS_YV4w

![image](https://github.com/user-attachments/assets/23c6d8de-3536-4b0d-8e37-c2bda902012c)

**Reply by allenjas** [reply]
Yep - DrErikson's screenshot is exactly what some of my students see:
![Error message](https://github.com/user-attachments/assets/20ec0b72-5eed-430e-8b04-074a7bde4581)

**Reply by DrErickson** [reply]
I figured out what was happening.  Check the GitHub Team plan under "Billing and Plans" on your organization:

https://github.com/settings/billing/summary

When I was re-verifying my educator status, GitHub revoked my benefits.  As a result, I had over 520 seats in my team and could not add any more students (it counts ALL students who ever checked out an assignment). I put my plan down to the "Free" tier, and it at least let my students check out the assignments.

A more descriptive error message may be beneficial in this case.

I'm hoping my benefits get re-instated again soon.

**Reply by katyhuff** [reply]
I am having the same issue today.

Classroom: https://classroom.github.com/classrooms/35508865-npre560-2024
Assignment: https://classroom.github.com/classrooms/35508865-npre560-2024/assignments/560-cp1
Assignment Invite Link: https://classroom.github.com/a/x_TQOgcw

**Reply by katyhuff** [reply]
As an update, I should note that my repo wasn't previously a template, so I changed it to a template. 
That didn't work, so I've just tried creating a brand new assignment that points to that same public template repo as the starter code.
**This second action seems to have worked,** so I'm now going to replace all of the assignments in this course by repeating that action.

**Reply by scerruti** [reply]
I am having the same issue. The link provided results in a 404. I re-reused the assignment from my sandbox and the new version has the same issue.

> Try and ([update](https://classroom.github.com/assignment-invitations/55aa62f5891adfacb8c8fa673c54fa5f/sync)) the assignment. If this problem persists, please have the Classroom admin contact support.

The next assignment in the course is working correctly.

Edit: I recreated the assignment from scratch and the new assignment is working.

**Comment by twennekers** [comment]
Hi

Thanks for dealing with this so quickly.

Unfortunately, the issue has not been resolved for me.

I tried yesterday and today to setup a student repository by reaccepting the assignment on a "student" test account.
The system tries to set the repository up correctly and asks to wait.
However, when I reload, the same out of sync error occurs.
There is a link on the error page to "update" the assignment which leads to a 404 error, page not found.

When I created the assignment a month or so ago I went through the creation process which worked at that time.
The repository is here:
https://github.com/Plymouth-University/comp1000-referral-2023-24-twennekers

Regards
Thomas

**Reply by g-simmons** [reply]
Same issue for me this week

**Comment by alanarnholt** [comment]
I am having the same issue as reported by VictorNorman.  

There was an error accepting the assignment. GitHub repository could not be created. The visibility of the starter code is out of sync with Classroom. Your instructor may need to re-create the assignment or change the visibility of the starter code repository.. Try and ([update](https://classroom.github.com/assignment-invitations/94a47ce862e71cf3103872f2c9718505/sync)) the assignment. If this problem persists, please have the Classroom admin contact support.

I recreated the assignment as suggested in the message and the error persisted.  Next I changed the starter repo locations....also to fail.

Suggestions most appreciated.

@alanarnholt

**Comment by jahagirdar** [comment]
I too am facing this issue for the assignment at https://classroom.github.com/a/D3N4nEj7
I have been using this repo for assignments for a few years now and have  464 repos in this org. Is this a problem?

**Comment by mshi311** [comment]
I'm having the same issue as well. I had one student who "left" the repository and then couldn't rejoin. I deleted her original repository from the github classroom (which was still there after she left) and had her try to accept again.  Now she is getting the following error when trying to access the document.

<img width="507" alt="Screen Shot 2024-10-09 at 10 18 22 PM" src="https://github.com/user-attachments/assets/609521fa-92d2-426c-ba18-4f024898fc93">
